onnx/defs/schema.cc

```diff
@@ -133,9 +113,30 @@ void OpSchema::Verify(const NodeProto& node) const {
           " is marked single but has an empty string in the graph");
     }
   }
+
   for (int out_idx = 0; out_idx < node.output_size(); ++out_idx) {
-    if (node.output(out_idx).empty()) {
-      fail_check("Output ", out_idx, " has an empty string in the graph");
+    if (out_idx >= outputs_.size()) {
+        if (outputs_.size() > 0 && Variadic == outputs_.back().GetOption()) {
+            // The last output formal parameter should be variadic.
+            break;
+        }
+        else {
+            fail_check(
+                "Node (",
+                node.name(),
+                ") has more outputs (",
+                node.output_size(),
+                ") than declared (",
+                outputs_.size(),
+                ") in op definition.");
+        }
+    }
+
+    if (node.output(out_idx).empty() && (Single == outputs_[out_idx].GetOption())) {
```

Comment:
so do we allow optional outputs now?

Reply:
yes, we do. In our current operators, BatchNormalization is the only one case to have optional outputs.